internal/jobexecutor/job_executor.go

```diff
@@ -131,7 +131,7 @@ func (e *JobExecutor) execute(ctx context.Context) (res *jobExecutorResult) {
 			)
 
 			res = &jobExecutorResult{
-				PanicTrace: string(debug.Stack()),
+				PanicTrace: captureStackTraceSkipFrames(4),
```

Comment:
Oof, it's good there's a test for this because it'd be pretty shaky otherwise. The raw number "4" probably merits a comment.